src/Tools/Source/CompilerGeneratorTools/Source/CSharpErrorFactsGenerator/ErrorGenerator.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// We only build the Source Generator in the netstandard target
```

Comment:
why?

Reply:
@CyrusNajmabadi We build this as an executable (.NET 7) which is used by `eng/generate-compiler-code.cmd`, and as a source generator (.NET Standard 2.0).

When compiling for net7.0, we don't need this as a source generator because generators should target netstandard2.0.

and similarly, when compiling for netstandard2.0, we don't need a Main entrypoint.

Reply:
Note: this follows the same pattern as https://github.com/dotnet/roslyn/commit/875e10444fad7ad52adc5544081b2be6b939191a (the comments are copy/pasted)